addons/point_of_sale/static/src/js/Popups/ClosePosPopup.js

```diff
@@ -152,16 +152,14 @@ class ClosePosPopup extends AbstractAwaitablePopup {
                 }
                 window.location = "/web#action=point_of_sale.action_client_pos_menu";
             } catch (error) {
-                const iError = identifyError(error);
-                if (
-                    iError instanceof ConnectionLostError ||
-                    iError instanceof ConnectionAbortedError
-                ) {
-                    await this.showPopup("ErrorPopup", {
-                        title: this.env._t("Network Error"),
-                        body: this.env._t("Cannot close the session when offline."),
-                    });
+                if (identifyError(error) instanceof ConnectionLostError) {
+                    // Cannot redirect to backend when offline, let error handlers show the offline popup
+                    // FIXME POSREF: doing this means closing again when online will redo the beginning of the method
+                    // although it's impossible to close again because this.closeSessionClicked isn't reset to false
+                    // The application state is corrupted.
```

Comment:
It would, I was just writing down some notes with regard with the way errors are handled in this flow, we should probably make it more robust but it was not the purpose of this PR.

Reply:
I just got what you meant, if the first two calls fails, it is handled outside and thus the `closeSessionClicked` will never be changed. I thought you were only talking about the last call. I thought you broke something (since I saw you throwing the error) without a finally block but you can't break something that's already broken :sweat_smile: 
So yeah sure, we can do that later.